test/common/upstream/health_checker_impl_test.cc

```diff
@@ -129,6 +129,21 @@ class HttpHealthCheckerImplTest : public testing::Test {
   HttpHealthCheckerImplTest()
       : cluster_(new NiceMock<MockClusterMockPrioritySet>()),
         event_logger_(new MockHealthCheckEventLogger()) {}
+  template <typename Config> void http_health_check_helper(Config conf, bool addHostCheck = true) {
+    if (addHostCheck == true) {
+      health_checker_ = std::make_shared<TestHttpHealthCheckerImpl>(
+          *cluster_, conf, dispatcher_, runtime_, random_,
+          HealthCheckEventLoggerPtr(event_logger_));
+      health_checker_->addHostCheckCompleteCb(
+          [this](HostSharedPtr host, HealthTransition changed_state) -> void {
+            onHostStatus(host, changed_state);
+          });
+    } else {
```

Comment:
actually I'd recommend splitting this into two functions:
  `allocHealthChecker();`
and
  `addCompletionCallback()`
thus avoiding the extra arg and, I think, making call-sites clearer.